src/features/todos.ts

```diff
@@ -1,8 +1,43 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
+import { getTodos } from '../api';
+
+type TodosState = {
+  todos: Todo[];
+  loading: boolean;
+  error: string;
+};
+
+const initialState: TodosState = {
+  todos: [],
+  loading: false,
+  error: '',
+};
+
+export const init = createAsyncThunk('todos/fetch', () => {
+  return getTodos();
+});
```

Comment:
Consider adding error handling within the `getTodos` function or within the `init` thunk to provide more detailed error messages or to handle specific error cases.